### PR TITLE
Add `server.prepareTransaction`

### PR DESCRIPTION
**Pull request by paulbellamy**
Adds a `server.prepareTransaction` method, which calls `simulateTransaction` then adds the resulting footprint into the transaction.

Depends on https://github.com/stellar/soroban-tools/pull/366

Fixes https://github.com/stellar/js-soroban-client/issues/39

### CODE REVIEW

File: src/soroban_rpc.ts [hunk 1]
```diff
@@ -39,6 +39,17 @@ export namespace SorobanRpc {
...
+  /* Response for jsonrpc method `getNetwork`
+   */
+  export interface GetNetworkResponse {
+    friendbotUrl?: string;
+    latestLedgerHash: string;
+    latestLedgerSeq: number;
+    latestLedgerTime: string;
+    passphrase: string;
+    protocolVersion: string;
+  }
```
Comment (Shaptic) [inline src/soroban_rpc.ts:51]: Can we reconcile this with the return values in [the doc](https://docs.google.com/document/d/1TZUDgo_3zPz7TiPMMHVW_mtogjLyPL0plvzGMsxSz6A/edit#heading=h.3zrwisrq3rg3)? It's hard to know where the source of truth is when reviewing fields/types/etc.

File: src/server.ts [hunk 2]
```diff
@@ -329,6 +342,66 @@ export class Server {
...
+   *
+   * @param {Transaction | FeeBumpTransaction} transaction - The transaction to
+   *    prepare. It should include exactly one operation, which must be a
+   *    {@link InvokeHostFunctionOp}. Any provided footprint will be overwritten.
+   * @returns {Promise<Transaction | FeeBumpTransaction>} Returns a copy of the
+   *    transaction, with the expected ledger footprint added.
+   */
+  public async prepareTransaction(
+    transaction: Transaction | FeeBumpTransaction,
+    networkPassphrase?: string,
```
Comment (Shaptic) [inline src/server.ts:394]: not documented? we should clarify that this will be an `async` call if this isn't passed

File: src/server.ts [hunk 3]
```diff
@@ -271,6 +270,20 @@ export class Server {
     });
   }
 
+  /**
+   * Fetches metadata about the network which Soroban-RPC is connected to.
+   *
+   * @example
+   * server.getNetwork();
```
Comment (Shaptic) [inline src/server.ts:277]: lol

**Review by Shaptic — APPROVED** [review]
LGTM, just some small doc notes :+1:

File: src/soroban_rpc.ts (hunk 1 above)
Comment (paulbellamy) [reply src/soroban_rpc.ts:51]: oh duh. Yeah. This response is wrong, sorry!

File: src/server.ts (hunk 3 above)
Comment (paulbellamy) [reply src/server.ts:277]: I mean... I'll add some console.logs of the outputs for this example

**Pull request merged**